src/core/thread/mle_router.cpp

```diff
@@ -264,11 +264,6 @@ ThreadError MleRouter::HandleDetachStart(void)
         mRouters[i].mState = Neighbor::kStateInvalid;
     }
 
```

Comment:
should we distinguish the reasons for DetachStart when deciding whether to keep child link or not?
what I want to say is that if DetachStart is caused by SetRouterRoleEnabled(false), it might not be necessary to keep child link any more; 
One scenario I could think of now is that if Router become detach due to SetRouterRoleEnabled(false),  the device would not act as router again, but cli 'child list' would still list deprecated child info before timeout child-which might not be expected.